### PR TITLE
#1367: Added Recent Artists Pie Chart to ListeningHistory

### PR DESCRIPTION
**Pull request by alicemmartinez**
Have not written test cases, or completely implemented filtering algorithm for range of dates. I was hoping you could guide me to where I could write test cases for my added subcomponent?

### CODE REVIEW

**Comment by nukeop** [comment]
- You can try moving this component to the `ui` package, and creating a story for it (using storybook). Then we could see what it looks like in various scenarios easily.
- In that package, there's an easy setup for snapshot tests (look for a function called `makeSnapshotTest`). If the chart is rendered in svg that's all we're going to need. You can also try adding more involved unit tests for any edge cases you can think of.
- If you do it that way, then this component can be imported into the `app` package, in the `ListeningHistoryView` component, and fed data from there. No need IMO to write any additional tests there. Just update existing snapshots (`jest -u`).

File: packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  '#33FFCC', '#66994D', '#B366CC', '#4D8000', '#B33300', '#CC80CC',
+  '#66664D', '#991AFF', '#E666FF', '#4DB3FF', '#1AB399', '#E666B3',
+  '#33991A', '#CC9999', '#B3B31A', '#00E680', '#4D8066', '#809980',
+  '#E6FF80', '#1AFF33', '#999933', '#FF3380', '#CCCC00', '#66E64D',
+  '#4D80CC', '#9900B3', '#E64D66', '#4DB380', '#FF4D4D', '#99E6E6',
+  '#6666FF'
+];
+
+// Function to get the top N items from the data based on a given key
+const getTopN = (data: any[], key: string, n: number): any[] => {
```
Comment (nukeop) [inline packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx:25]: Make this function better by making it generic (`<T>`), instead if using `any`. That way we can ensure type safety for its input and output.

File: packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  return data?.sort((a, b) => b[key] - a[key]).slice(0, n);
+};
+
+export type PieChartTopArtists = {
+    data: any[];
+}
+
+const timeframes = ['Last 7 days', 'Last 30 days', 'Last 90 days', 'Last 180 days', 'Last 365 days', 'All time'];
```
Comment (nukeop) [inline packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx:33]: Nuclear is translated into more than 30 languages. So we should never use hardcoded strings like these, and instead use `react-i18next`. Try looking for the `useTranslation` hook to see how it's used.

If you decide to move this component to the `ui` package, you can simply feed it strings it needs to render its interface. For an example of how it can be done, you can look for `TrackPopupStrings`. The idea is that the component in `ui` takes a list of texts it's going to render, and in the `app` package, when we want to use that component, we look up the translated texts using the `useTranslation` hook. If you're unsure how to do this, let me know.

Comment (nukeop) [reply packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx:33]: And the same comment applies to other instances of raw text, like `Timeframe`, etc.

File: packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          >
+            {timeframes.map((tf) => (
+              <option key={tf} value={tf}>
+                {tf}
+              </option>
+            ))}
+          </select>
+        </label>
+
+        <label style={{ marginLeft: '20px' }}>
```
Comment (nukeop) [inline packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx:59]: Instead of using the `style` attribute, try using scss files and classes, it separates the styles from the markup.

File: packages/app/app/components/ListeningHistoryView/testSuite.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
```
Comment (nukeop) [inline packages/app/app/components/ListeningHistoryView/testSuite.js:1]: The convention for tests is to use `<container>.test.ts`.

File: packages/app/app/components/ListeningHistoryView/testSuite.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+const mockData = [
+  { name: 'Artist A', count: 50 },
+  { name: 'Artist B', count: 40 },
+  { name: 'Artist C', count: 30 },
+  { name: 'Artist D', count: 20 },
+  { name: 'Artist E', count: 10 }
+];
+
+describe('PieChartTopArtists', () => {
+  test('renders without crashing', () => {
```
Comment (nukeop) [inline packages/app/app/components/ListeningHistoryView/testSuite.js:14]: This can be replaced with a snapshot test, which gives us the same information, and also records what the component looks like in the HTML.
